File: LeetCode/LeetCode Practice Problems/589- N-ary Tree Preorder Traversal.cpp
class Solution {
public:
    vector<int> ans;
    
    void traverse(Node* root) {
        if(root == NULL) 
            return;
        
        ans.push_back(root->val);
        
        for(auto i: root->children) {
            traverse(i);
        }
    }
    
    vector<int> preorder(Node* root) {
        traverse(root);
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/589-N-ary Tree Preorder Traversal.java
class Solution {
    List<Integer> ans = new ArrayList();
    
    public List<Integer> preorder(Node root) {
        if(root != null) {
            ans.add(root.val);
            for(Node child: root.children) {
                preorder(child);
            }
        }
        return ans;
    }
}
